docs/architecture/adr-102-rpc-companion.md

```diff
@@ -0,0 +1,341 @@
+# ADR-102: RPC Companion
+
+## Changelog
+
+- 2022-03-27: First draft (@andynog)
+
+## Status
+
+Accepted | Rejected | Deprecated | Superseded by
+
+## Context
```

Comment:
Is it worth mentioning that one advantage of this setup is that the full node can turn off indexing (which is a known source of contention and performance bottleneck)?

Reply:
added, thanks